src/flag_gems/utils/libentry.py

```diff
@@ -1,12 +1,130 @@
 import inspect
+import sqlite3
 import threading
+import weakref
+from typing import Dict
 
 import torch
 import triton
 
+from .code_cache import config_cache_dir
+
 DEVICE_COUNT = torch.cuda.device_count()
 
 
+class LibTuner(triton.runtime.Autotuner):
+    def __init__(
+        self,
+        fn,
+        arg_names,
+        configs,
+        key,
+        reset_to_zero,
+        restore_value,
+        pre_hook=None,
+        post_hook=None,
+        prune_configs_by: Dict = None,
+        warmup=25,
+        rep=100,
+        use_cuda_graph=False,
+    ):
+        super().__init__(
+            fn,
+            arg_names,
+            configs,
+            key,
+            reset_to_zero,
+            restore_value,
+            pre_hook,
+            post_hook,
+            prune_configs_by,
+            warmup,
+            rep,
+            use_cuda_graph,
+        )
+        self.__name__ = self.base_fn.__name__
+        self.cache_dir = config_cache_dir() / f"{self.__name__}.db"
+        self.preload()
+        weakref.finalize(self, self.store)
+
+    def getvalue(self, key):
+        try:
+            return eval(key)
+        except Exception:
+            return getattr(torch, key.split(".")[1][:-1])
+
+    def preload(self):
+        connect = sqlite3.connect(self.cache_dir)
+        c = connect.cursor()
+        c.execute(f"CREATE TABLE IF NOT EXISTS {self.__name__} (key TEXT, config TEXT)")
+        cursor = c.execute(f"SELECT key, config from {self.__name__}")
+
+        for row in cursor:
+            key_str, config_str = row
+            key = (self.getvalue(k) for k in key_str[1:-1].split(", "))
+
+            cfg_ls = [item.split(": ") for item in config_str.split(", ")]
+            config = triton.Config({})
+            for k, v in cfg_ls[:-4]:
+                config.kwargs[k] = eval(v)
+            config.num_warps = int(cfg_ls[-4][1])
+            config.num_ctas = int(cfg_ls[-3][1])
+            config.num_stages = int(cfg_ls[-2][1])
+            config.enable_fp_fusion = eval(cfg_ls[-1][1])
+
+            self.cache[key] = config
+
+        connect.close()
```

Comment:
Probably we could leave the connection open and shared because we need to fetch stored params for all gems ops. A try catch block is possibly necessary for external IO operations.

Reply:
But I think its fine at this stage. Let's improve it later.

Reply:
there is not always a explicit exit of flag_gems. I'm not sure what time to close the connection if keep it open.